packages/aws-cdk/lib/api/hotswap/lambda-functions.ts

```diff
@@ -281,3 +295,55 @@ class LambdaFunctionHotswapOperation implements HotswapOperation {
     return Promise.all(operations);
   }
 }
+
+/**
+ * Compress a string as a file, returning a promise for the zip buffer
+ * https://github.com/archiverjs/node-archiver/issues/342
+ */
+export function zipString(fileName: string, rawString: string): Promise<Buffer> {
+  return new Promise((resolve, reject) => {
+    const buffs: Buffer[] = [];
+
+    const converter = new Writable();
+
+    converter._write = (chunk: Buffer, _: string, cb: () => void) => {
+      buffs.push(chunk);
+      process.nextTick(cb);
+    };
+
+    converter.on('finish', () => {
+      resolve(Buffer.concat(buffs));
+    });
+
+    const archive = archiver('zip');
+
+    archive.on('error', (err) => {
+      reject(err);
+    });
+
+    archive.pipe(converter);
+
+    archive.append(rawString, {
+      name: fileName,
+      date: new Date('1980-01-01T00:00:00.000Z'), // Add date to make resulting zip file deterministic
+    });
+
+    void archive.finalize();
+  });
+}
+
+/**
+ * Get file extension from Lambda runtime string.
+ * We use this extension to create a deployment package from Lambda inline code.
+ */
+function getExtensionFromRuntime(runtime: string): string {
+  if (runtime.startsWith('node')) {
+    return 'js';
+  }
+  if (runtime.startsWith('python')) {
+    return 'py';
+  }
+  // Currently inline code only supports Node.js and Python, ignoring other runtimes.
+  // https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-lambda-function-code.html#aws-properties-lambda-function-code-properties
+  return 'js';
```

Comment:
```suggestion
  if (runtime.startsWith('node')) {
    return 'js';
  }
  if (runtime.startsWith('python')) {
    return 'py';
  }
  // Currently inline code only supports Node.js and Python, ignoring other runtimes.
  // https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-lambda-function-code.html#aws-properties-lambda-function-code-properties
  throw new Error(`runtime ${runtime} is unsupported, only node.js and python runtimes are currently supported.`);
  
  @skinny85 thoughts on this change?
```

Reply:
Definitely not `Error`, because that will just error the entire deployment. We need to throw a `CfnEvaluationException`, so that a full deployment can be done instead.

Reply:
Yes, I like Calvin's suggestion. Throw a `CfnEvaluationException` here, saying you couldn't determine the file extension for the runtime `runtime`.

This will be caught by [this code](https://github.com/aws/aws-cdk/blob/c7c51debabeb3ec1de4f4c2dc986dfdfac27756b/packages/aws-cdk/lib/api/deploy-stack.ts#L259-L263), and then a full deployment will happen, instead of a hotswap, which is the behavior we want.